test/helpers/PlayerInteractionWrapper.js

```diff
@@ -309,6 +317,15 @@ class PlayerInteractionWrapper {
         });
     }
 
+    attachOpponentOwnedUpgrades(opponentOwnedUpgrades = []) {
+        opponentOwnedUpgrades.forEach((upgrade) => {
+            const upgradeCard = this.findCardByName(upgrade.card, 'any', 'opponent');
```

Comment:
I worry that using `findCardByName` in a lot of places will cause things to break if there are multiple cards in the field with the same name. Any reason we're using strings here instead of just passing around explicit references to the card objects like with some of the other test methods?

Reply:
This is because the opponentOwnedUpgrades array is not actual cards but is instead just the JSON representation of the setup info. We have to map that to the actual card using findCardByName